Sources/Workspace/Destination.swift

```diff
@@ -180,33 +181,35 @@ public struct Destination: Encodable, Equatable {
 }
 
 extension Destination {
-
     /// Load a Destination description from a JSON representation from disk.
     public init(fromFile path: AbsolutePath, fileSystem: FileSystem = localFileSystem) throws {
-        let json = try JSON(bytes: fileSystem.readFileContents(path))
-        try self.init(json: json)
-    }
-}
-
-extension Destination: JSONMappable {
-
-    /// The current schema version.
-    static let schemaVersion = 1
-
-    public init(json: JSON) throws {
-
+        let decoder = JSONDecoder.makeWithDefaults()
+        let version = try decoder.decode(path: path, fileSystem: fileSystem, as: VersionInfo.self)
         // Check schema version.
-        guard try json.get("version") == Destination.schemaVersion else {
+        guard version.version == 1 else {
             throw DestinationError.invalidSchemaVersion
         }
-
+        let destination = try decoder.decode(path: path, fileSystem: fileSystem, as: DestinationInfo.self)
         try self.init(
-            target: Triple(json.get("target")),
-            sdk: AbsolutePath(json.get("sdk")),
-            binDir: AbsolutePath(json.get("toolchain-bin-dir")),
-            extraCCFlags: json.get("extra-cc-flags"),
-            extraSwiftCFlags: json.get("extra-swiftc-flags"),
-            extraCPPFlags: json.get("extra-cpp-flags")
+            target: destination.target.map{ try Triple($0) },
+            sdk: destination.sdk,
+            binDir: destination.binDir,
+            extraCCFlags: destination.extraCCFlags,
+            extraSwiftCFlags: destination.extraSwiftCFlags,
+            extraCPPFlags: destination.extraCCFlags
         )
     }
 }
+
+fileprivate struct VersionInfo: Codable {
+    let version: Int
+}
+
+fileprivate struct DestinationInfo: Codable {
+    let target: String?
+    let sdk: AbsolutePath?
+    let binDir: AbsolutePath
```

Comment:
I just noticed that this small change from this much larger pull changes the format for destination JSON files, rather than what's documented [here](https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#additional-flags) and [here](https://github.com/apple/swift-package-manager/blob/main/Utilities/build_ubuntu_cross_compilation_toolchain#L240).

@tomerd, should I update the docs and example for the new format or should we stick with the old format? I'll submit a pull for either way you want to go.

Reply:
@buttaface great catch, would appreciate such help

Reply:
actually, I think we can also consider the reverse it the old name is more correct - ie we could also rename / change the decoding to accept the old name, as the rename was more cleanup code wise

Reply:
iow the goal here was not to change the format / names, but an internal cleanup of the decoding to use Codable, the name change was an oversight, so happy to change back to the name if preferable

Reply:
def this is missing a test tho :D

Reply:
OK, I will change it back and add a test.

Reply:
Submitted two fixes in #3793, will submit the test later.